File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int slow = 0, fast = 0, itr = 0;
        
        while(true){
            
            slow = nums[slow];
            fast = nums[nums[fast]];
            
            if(slow == fast){
                break;
            }
        }
        
        while(true){
            itr = nums[itr];
            slow = nums[slow];
            
            if(slow == itr){
                return slow;
            }
        }
    }
};

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        slow = fast = itr = 0
        
        #detect the dupe
        while True:
            slow = nums[slow]
            fast = nums[nums[fast]]
            
            if slow == fast:
                break
        #find the dupe
        while True:
            itr = nums[itr]
            slow = nums[slow]
            
            if slow == itr:
                return slow
        